src/models/simulation.jl

```diff
@@ -51,14 +55,25 @@ callbacks and performs the first two initial time steps to spin up the
 leapfrogging scheme."""
 function initialize!(
     simulation::AbstractSimulation;
-    period::Period = Day(10),
+    period::Period = DEFAULT_PERIOD,
+    steps::Int = DEFAULT_TIMESTEPS,
     output::Bool = false,
 )
     progn, diagn, model = unpack(simulation)
 
     # SET THE CLOCK
-    set_period!(progn.clock, period)                            # set how long to integrate for
-    initialize!(progn.clock, simulation.model.time_stepping)    # store the start date, reset counter
+    (; clock) = progn
```

Comment:
To me it seems like the cleaner solution to put all of this in `initialize!(::Clock)`

Reply:
Yes, just did that with the latest commit. Removed the `set_period!` function+methods and it's all in `initialize!` now with several methods for that